### PR TITLE
Use ZFP for compressing discrete trajectories

### PR DESCRIPTION
**Pull request by pleroy**
This seems to buy us a factor 4 in size, while complying with the downsampling tolerance of 10 m.

Helps with #2400.

### CODE REVIEW

File: physics/discrete_trajectory_body.hpp [hunk 1]
```diff
@@ -494,12 +503,64 @@ void DiscreteTrajectory<Frame>::WriteSubTreeToMessage(
...
+
+  // Times are exact.
+  ZfpCompressor time_compressor(0);
+  // Lengths are approximated to the downsampling tolerance if downsampling is
+  // enabled, otherwise they are exact.
+  Length const length_tolerance =
+      downsampling_.has_value() ? downsampling_->tolerance() : Length();
+  ZfpCompressor length_compressor(length_tolerance / Metre);
+  // Speeds are approximated based on the length tolerance and the average step
+  // in the timeline.
```
Comment (eggrobin) [inline physics/discrete_trajectory_body.hpp]: Let us not average things over the timeline; this is meaningless (the timeline may behave completely differently at different times) and it might result in weird edge cases that are hard to find and test, e.g., an interplanetary trajectory (large Δt, thus requiring a small speed tolerance to keep the error of the Hermite interpolation in check) getting mangled by a long time spent in low orbit, or even just by a lot of dense intervals (low Δt, 10 s for dense intervals, driving the mean Δt down).

Use the max Δt.

Comment (pleroy) [reply physics/discrete_trajectory_body.hpp]: Done, PTAL.

File: serialization/physics.proto [hunk 2]
```diff
@@ -89,6 +89,12 @@ message DiscreteTrajectory {
   repeated int32 fork_position = 3;
   // Added in 陈景润.
   optional Downsampling downsampling = 4;
+
+  // ZFP support added in Frobenius.
+  optional int32 zfp_codec_version = 7;
+  optional int32 zfp_library_version = 8;
+  optional bytes zfp_timeline = 5;
+  optional int32 zfp_timeline_size = 6;
```
Comment (eggrobin) [inline serialization/physics.proto]: These should be wrapped in a message, since they are really optional and none or all should be present.

Comment (pleroy) [reply serialization/physics.proto]: Done, PTAL.
